devtools/src/arcs-hcr-list.js

```diff
@@ -0,0 +1,314 @@
+              const spec = m.pecMsgBody.spec;
+
+              this.allParticlesInArc.push({id: particleId, spec: spec});
+              this._displayLog(particleId, spec);
+            }
+          }
+          break;
+        }
+      }
+    }
+  }
+
+  _forceReload() {
+    this.reloadableParticles.forEach(particle => {
+      this._updateLog(particle.implFile);
+      this.arcIdsToKeepFresh.forEach(id => {
+        this.send({
+            messageType: 'particle-reload',
+            messageBody: particle.implFile,
+            arcId: id
+        });
+      });
+    });
+  }
+
+  _displayLog(particleId, spec) {
+    if (this.pathToTimeLog.has(spec.implFile)) {
+      const particle = {
+        name: spec.name,
+        id: particleId,
+        implFile: spec.implFile,
+        time: this.pathToTimeLog.get(spec.implFile)
+      };
+      this.reloadableParticles.push(particle);
+    }
+  }
+
+  _updateLog(filePath) {
+    const time = this._getCurrentTime();
+    if (!this.pathToTimeLog.has(filePath)) {
+      this.pathToTimeLog.set(filePath, []);
+    }
+    const timeLog = this.pathToTimeLog.get(filePath);
+    if (!timeLog.includes(time)) timeLog.push(time);
+
+    [...this.reloadableParticles].forEach(particle => {
+      if (particle.implFile == filePath) {
+        const explorer =  this.shadowRoot.querySelector(`object-explorer[log-id="${filePath}"]`);
+        if (explorer) {
+          explorer.refresh();
+          explorer.flash();
+        }
+      }
+    });
+  }
+
+  _loadServer() {
+    if (this.$.toggle.checked) {
+      this.$.toggle.checked = false;
+
+      this.websocket = new WebSocket('ws://localhost:8888');
+      this.websocket.onopen = e => {
+        this.$.toggle.checked = true;
+
+        this.websocket.onmessage = msg => {
+          const input = (`${msg.data}`).split(' ');
+          this._updateLog(input[1]);
+          switch (input[0]) {
+            case 'watch':
+              this.reloadableParticles = [];
+              this.allParticlesInArc.forEach(p => {
+                this._displayLog(p.id, p.spec);
+              });
+              break;
+            case 'reload':
+              this.arcIdsToKeepFresh.forEach(id => {
+                this.send({
+                    messageType: 'particle-reload',
+                    messageBody: input[1],
+                    arcId: id
+                });
+              });
+              break;
+          }
+        };
+        this.websocket.send(JSON.stringify([...this.particleSources]));
+      };
+      this.websocket.onerror = e => {
+        alert('No ALDS connection found');
+      };
+
+    } else {
+      if (this.ws != null) {
+        alert('Something went wrong with hot code reload');
+      }
+      this.websocket.close();
+      this.websocket = null;
+      this.logging = new Map();
+      this.reloadableParticles = [];
+    }
+  }
+
+  _getCurrentTime() {
+    const ts = Date.now();
```

Comment:
`new Date(ts).getTime()`will give you ts back.

it's enough to do:
```
_getCurrentTime() {
  return formatTime(Date.now());
}
```

Reply:
Changed!